File: app/src/main/java/com/kapanen/hearthstoneassessment/model/Card.kt
package com.kapanen.hearthstoneassessment.model

import android.os.Parcelable
import com.kapanen.hearthstoneassessment.data.CardType
import kotlinx.parcelize.Parcelize

/**
 *
 * @param cardId
 * @param cardType
 * @param name
 * @param cardSet
 * @param type
 * @param rarity
 * @param cost
 * @param attack
 * @param health
 * @param htmlText
 * @param flavor
 * @param artist
 * @param elite
 * @param playerClass
 * @param multiClassGroup
 * @param img
 * @param imgGold
 * @param mechanics
 */
@Parcelize
data class Card(
    val cardId: String,
    val cardType: CardType? = null,
    val name: String? = null,
    val cardSet: String? = null,
    val type: String? = null,
    val faction: String? = null,
    val rarity: String? = null,
    val cost: Int? = null,
    val attack: Int? = null,
    val health: Int? = null,
    val htmlText: String? = null,
    val flavor: String? = null,
    val artist: String? = null,
    val collectible: Boolean? = null,
    val elite: Boolean? = null,
    val playerClass: String? = null,
    val multiClassGroup: String? = null,
    val img: String? = null,
    val imgGold: String? = null,
    val mechanics: List<Mechanic>? = null,
    val isFavorite: Boolean = false
) : Parcelable

File: Hearthstone/Hearthstone/Models/Card.swift
//
//  Card.swift
//  Hearthstone
//
//  Created by Stavros Tsikinas on 26/7/22.
//

import Foundation

// For the purpose of the DEMO we make the assumption that we know the Card Categories beforehand (if API requests, the setup would be different)
struct CardsResponse: Codable {
    var Basic: [Card] = Array()
    var Classic: [Card] = Array()
    var Promo: [Card] = Array()
    var HallofFame: [Card] = Array()
    var Naxxramas: [Card] = Array()
    var GoblinsvsGnomes: [Card] = Array()
    var BlackrockMountain: [Card] = Array()
    var TheGrandTournament: [Card] = Array()
    var TheLeagueofExplorers: [Card] = Array()
    var WhispersoftheOldGods: [Card] = Array()
    var OneNightinKarazhan: [Card] = Array()
    var MeanStreetsofGadgetzan: [Card] = Array()
    var JourneytoUnGoro: [Card] = Array()
    var TavernBrawl: [Card] = Array()
    var HeroSkins: [Card] = Array()
    var Missions: [Card] = Array()
    var Credits: [Card] = Array()
    var System: [Card] = Array()
    var Debug: [Card] = Array()
}

struct Card: Codable {

    var cardId: String?
    var name: String?
    var cardSet: String?
    var type: String?
    var rarity: String?
    var cost: Int?
    var attack: Int?
    var health: Int?
    var text: String?
    var flavor: String?
    var artist: String?
    var collectible: Bool?
    var elite: Bool?
    var playerClass: String?
    var multiClassGroup: String?
    var classes: [String]? = Array()
    var img: String?
    var imgGold: String?
    var locale: String?
    var mechanics: [[String: String]]? = Array()
    
}
